File: frontend/types/analytics.ts
export interface MetricCount {
  name: string;
  count: number;
}

export interface DateMetricCount {
  date: string;
  count: number;
}

export interface MaintenancePerformance {
  maintenance_id: string;
  maintenance_name: string;
  completed_count: number;
  active_count: number;
}

export interface AnalyticsSummary {
  total_issues: number;
  open_issues: number;
  verified_issues: number;
  assigned_issues: number;
  in_progress_issues: number;
  resolved_issues: number;
  rejected_issues: number;
  high_priority_issues: number;
  critical_issues: number;
  overdue_issues: number;
  avg_resolution_hours: number;
}

export interface FullAnalyticsResponse {
  summary: AnalyticsSummary;
  by_status: MetricCount[];
  by_category: MetricCount[];
  by_location: MetricCount[];
  by_priority: MetricCount[];
  reports_over_time: DateMetricCount[];
  resolved_vs_unresolved: MetricCount[];
  maintenance_performance: MaintenancePerformance[];
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class MetricCount(BaseModel):
    name: str
    count: int


class DateMetricCount(BaseModel):
    date: str
    count: int


class MaintenancePerformance(BaseModel):
    maintenance_id: str
    maintenance_name: str
    completed_count: int
    active_count: int


class AnalyticsSummary(BaseModel):
    total_issues: int
    open_issues: int
    verified_issues: int
    assigned_issues: int
    in_progress_issues: int
    resolved_issues: int
    rejected_issues: int
    high_priority_issues: int
    critical_issues: int
    overdue_issues: int
    avg_resolution_hours: float


class FullAnalyticsResponse(BaseModel):
    summary: AnalyticsSummary
    by_status: list[MetricCount]
    by_category: list[MetricCount]
    by_location: list[MetricCount]
    by_priority: list[MetricCount]
    reports_over_time: list[DateMetricCount]
    resolved_vs_unresolved: list[MetricCount]
    maintenance_performance: list[MaintenancePerformance]
